src/lein_monolith/config.clj

```diff
@@ -121,6 +121,19 @@
     :else nil))
 
 
+(defn- with-absolute-clean-targets
+  "Returns the given project map with its clean targets updated to use absolute paths
+   to work around https://github.com/technomancy/leiningen/issues/2707"
+  [{:keys [root clean-targets] :as project}]
+  (let [abs-target-fn (fn [target]
+                        (if (and (string? target)
+                                 (not (.isAbsolute (io/file target))))
+                          (str (io/file root target))
+                          target))
+        abs-clean-targets (with-meta (mapv abs-target-fn clean-targets) (meta clean-targets))]
+    (assoc project :clean-targets abs-clean-targets)))
```

Comment:
🤔 What if there are no clean targets specified? Should this guard against that?

Reply:
My understanding is that lein provides a default of `[:target-path]` here if no clean targets are specified:
https://github.com/amperity/lein-monolith/pull/99

🤔 Do you mean we should be guarding against someone unsetting this in a profile (or setting it explicitly to nil)?

Reply:
err sorry, i meant to link to https://codeberg.org/leiningen/leiningen/src/commit/144a1de7d68b5f06620f68f38fe53cf61ffabab2/leiningen-core/src/leiningen/core/project.clj#L226 🤦

Reply:
That seems reasonable to expect the value to be present, and if it was `nil` then an empty vector probably has the same behavior.